File: week-4/day-1/sum of extracted digits.py
def extract_second_last_digit(number):
    if number < 10:
        return None 
    number_str = str(number)
    return int(number_str[-2])


n1 = int(input())
n2 = int(input())

second_last_digit_n1 = extract_second_last_digit(n1)
second_last_digit_n2 = extract_second_last_digit(n2)

if second_last_digit_n1 is None or second_last_digit_n2 is None:
    print("Invalid number")
else:
    
    result = second_last_digit_n1 + second_last_digit_n2
    print(result)

File: week-4/day-2/Sum of extracted digits.cpp
#include <iostream>
using namespace std;

// Function to extract the second last digit
int extractSecondLastDigit(long long number) {
    // Check if the number has fewer than 2 digits
    if (number < 10) {
        return -1;  // Indicate invalid number
    }
    // Remove the last digit
    number /= 10;
    // Extract the new last digit which is the second last digit of the original number
    return number % 10;
}

int main() {
    long long num1, num2;

    // Take input from user
    cout << "Enter two numbers: ";
    cin >> num1 >> num2;

    // Extract the second last digit from each number
    int digit1 = extractSecondLastDigit(num1);
    int digit2 = extractSecondLastDigit(num2);

    // Check if either of the numbers was invalid
    if (digit1 == -1 || digit2 == -1) {
        cout << "Invalid number" << endl;
    } else {
        // Print the sum of the second last digits
        cout << (digit1 + digit2) << endl;
    }

    return 0;
}
